src/main/java/frc/robot/auto_manager/AutoNoteStep.java

```diff
@@ -51,7 +37,23 @@ private static Supplier<Optional<Translation2d>> noteIdToPose(int id) {
     return new AutoNoteStaged(id)::getPose;
   }
 
-  public AutoNoteStep(AutoNoteAction action, Supplier<Optional<Translation2d>>... notes) {
-    this(action, List.of(notes));
+  private static List<Supplier<Optional<Translation2d>>> noteIdsToSearchPoseSuppliers(
+      Integer... noteIds) {
+    List<Integer> list = List.of(noteIds);
+
+    return list.stream()
+        .map(
+            noteId -> {
+              // Explicit about types here since Java can't infer the return types
+              Supplier<Optional<Translation2d>> searchPoseSupplier =
+                  () -> AutoNoteStep.noteIdToPose(noteId).get();
```

Comment:
this is the same as the old code with two differences:
1. instead of type casting we just are explicit with variable + return types, which means no casting needed
2. a critical skill issue occurred while i was hacking together the old version. you will notice that it's hardcoded to look for note 0. that was why the code was crashing (since 0 is not a valid note ID)